GitUI/GitUIExtensions.cs

```diff
@@ -88,6 +88,15 @@ public static Task ViewChanges(this FileViewer diffViewer, string firstRevision,
         {
             return diffViewer.ViewPatch(() =>
             {
+                if (firstRevision == null && secondRevision.IsNotNullOrWhitespace())
```

Comment:
This `if` is inside async task of loading a patch. It should be outside of it. I created a PR https://github.com/gerhardol/gitextensions/pull/2 to show how I would see it done.

Also replaced `secondRevision` in `if`  by ensuring it is not empty.